### ISSUE TITLE
Documentation is hard to follow

### ISSUE THREAD

**Issue by CyborTronik** [b3b00/csly#168]
As example: While reading lexer section I see an example of enum. And no clue how to use it.
Only at the end of documentation or in examples you may find some of such info. Which makes it hard to follow.

Usually for a better experience, docs starts with a 'Getting Started' section which helps in few lines to do a sample. And this provides a common ground for more advanced details. And through documentation you learn how to enhance different pieces.

So good idea overall. So hard to use

**Comment by b3b00** [comment]
Hi,

I agree that the documentation is not a getting started. When you say "so good idea overall. so hard to use" what do you exactly mean ? 

I am not a big fan of the "getting started" docs, they only show the surface of a project and people may not really understand the whole thing.

You could maybe contribute with some documentation as you would have expected it. At least a draft....

**Comment by b3b00** [comment]
@CyborTronik , was you talking about the readme or the wiki doc ?
